lib/src/http/common/request.dart

```diff
@@ -415,16 +446,23 @@ abstract class CommonRequest extends Object
           !(response.status >= 200 && response.status < 300)) {
         throw new RequestException(method, this.uri, this, response);
       }
+
+      if (responseInterceptor != null) {
+        response = await responseInterceptor(finalizedRequest, response);
+      }
     } catch (e) {
       var error = e;
       if (!_done.isCompleted) {
         _done.complete();
       }
       cleanUp();
-      checkForCancellation(response: response);
       if (error is! RequestException) {
         error = new RequestException(method, this.uri, this, response, error);
       }
+      if (responseInterceptor != null) {
+        response = await responseInterceptor(finalizedRequest, response, error);
```

Comment:
It seems as if this `catch` block could be entered if there was an error encountered during `await responseInterceptor` on line 451, and then it would do it again on line 463. Is that a correct understanding? Should we prevent that situation?